Rebus/Encryption/EncryptMessagesOutgoingStep.cs

```diff
@@ -37,10 +37,11 @@ public async Task Process(OutgoingStepContext context, Func<Task> next)
 
             var headers = transportMessage.Headers.Clone();
             var bodyBytes = transportMessage.Body;
-            var encryptedData = _encryptor.Encrypt(bodyBytes);
+            var encryptedData = await _encryptor.Encrypt(bodyBytes);
 
             headers[EncryptionHeaders.ContentEncryption] = _encryptor.ContentEncryptionValue;
             headers[EncryptionHeaders.ContentInitializationVector] = Convert.ToBase64String(encryptedData.Iv);
+            headers[EncryptionHeaders.KeyId] = encryptedData.KeyId;
```

Comment:
I didn't dig deep enough to see - will empty headers be omitted? If not, I'd like to add a null check here

Reply:
>  will empty headers be omitted?

No they won't. So yeah, it would be nice to skip setting the header if the value is `null`.

Reply:
[Fix'd](https://github.com/rebus-org/Rebus/pull/927/commits/a2377065724fdc7178b3a4629cbbab6030cdb655#diff-cea16257463d1bb4ad07afbb6e6ec096491aa017766dba3642ac2bb69d8ebae0R47)